test/e2e/test_image.py

```diff
@@ -95,3 +96,177 @@ def test_create_with_invalid_image_file(self):
 
         self.post_image(IMAGE_POST_METADATA_DATA_REQUIRED_VALUES_ONLY, "invalid_image.jpg")
         self.check_post_image_failed_with_detail(422, "File given is not a valid image")
+
+
+# pylint:disable=fixme
+# TODO: Inherit from GetDSL when added
+class ListDSL(CreateDSL):
+    """Base class for list tests."""
+
+    _get_response_image: Response
+
+    def get_images(self, filters: Optional[dict] = None) -> None:
+        """
+        Gets a list of images with the given filters.
+
+        :param filters: Filters to use in the request.
+        """
+        self._get_response_image = self.test_client.get("/images", params=filters)
+
+    def check_get_images_success(self, expected_images_get_data: list[dict]) -> None:
+        """
+        Checks that a prior call to `get_images` gave a successful response with the expected data returned.
+
+        :param expected_images_get_data: List of dictionaries containing the expected image data as would
+            be required for an `ImageSchema`.
+        """
+        assert self._get_response_image.status_code == 200
+        assert self._get_response_image.json() == expected_images_get_data
+
+    def check_images_list_response_failed_with_message(self, status_code, expected_detail, obtained_detail):
+        """Checks the response of listing images failed as expected"""
+
+        assert self._get_response_image.status_code == status_code
+        assert obtained_detail == expected_detail
+
+    def post_test_images(self) -> list[dict]:
+        """
+        Posts three images. The first two have the same entity ID, and all of them have different titles and descriptions.
+
+        :return: List of dictionaries containing the expected item data returned from a get endpoint in
+                 the form of an `ImageSchema`.
+        """
+
+        entity_id_a, entity_id_b = (str(ObjectId()) for _ in range(2))
+
+        # First item
+        image_a_id = self.post_image({**IMAGE_POST_METADATA_DATA_ALL_VALUES, "entity_id": entity_id_a}, "image.jpg")
+
+        # Second item
+        image_b_id = self.post_image(
+            {
+                **IMAGE_POST_METADATA_DATA_ALL_VALUES,
+                "entity_id": entity_id_a,
+                "title": "Essay Title",
+                "description": "A research essay.",
+            },
```

Comment:
```suggestion
            {
                **IMAGE_POST_METADATA_DATA_REQUIRED_VALUES_ONLY,
                "entity_id": entity_id_a,
            },
```
Its nicer to use data from the mock data where possible to avoid repeatedly defining fields. The title and description doesn't really need to be different for each item with a single one we are already checking it gets used correctly as there is no specific logic to avoid duplicates etc, so C can probably be identical to A or B except the entity ID. Should mate the return values have less fields as well.